### ISSUE TITLE
JSX of Official Modules

### ISSUE THREAD

**Issue by pronetsweb** [elegantthemes/create-divi-extension#156]
We have several custom modules that extend / modify official modules. On the PHP side this is easy enough via inheritance:

`class PNI_GalleryWithLinks extends ET_Builder_Module_Gallery {`

However rendering the component in the visual builder is much more challenging as we would have to rewrite the JSX of the entire module. Is there any way to call / reuse the JSX code of an official module? For example, in our gallery with links example above, the gallery's appearance is pretty much identical to a normal gallery.

Advice / information would be appreciated.

Thanks!

**Comment by etstaging** [comment]
### QA Checklist
<strong>Reporter</strong>
- [ ] Helpful title used &nbsp;<img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
  * ***Helpful title format:*** `Scope(s) :: One Sentence To Describe The Issue`
  * ***Example:*** `VB :: Floating Toolbar :: Jumps To Top Of Page`
- [ ] Detailed description <img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
  * ***Description should usually be 150+ characters.***
- [ ] Steps to reproduce included <img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
- [ ] Support Ticket URL included
- [ ] Image and/or gif

<strong>Developer</strong>
- [ ] Steps to reproduce proven solved
- [ ] Test included and coverage acceptable and all Tests passing?
- [ ] PR description has layman's description of problem and solution and "Areas Affected" listed/explained?
- [ ] PR branch includes latest commits from master branch (at the time of checking)?
- [ ] Estimated points on issue
- [ ] Proper code documentation and conventions?
- [ ] Proper RTL appearance
- [ ] i18n used if applicable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated?

<strong>Peer Review</strong>
- [ ] PR branch includes latest commits from master branch (at the time of checking)?
- [ ] Steps to reproduce proven solved
- [ ] Affected areas are tested and ok
- [ ] Proper code documentation and conventions
- [ ] Proper RTL appearance
- [ ] i18n used if applicable
- [ ] Test(s) included and coverage acceptable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated?

<strong>Quality Assurance</strong>
- [ ] All tests passed on CircleCI.
- [ ] Steps to reproduce passed, the fix works.
- [ ] No errors in console or output 
- [ ] Affected areas are tested and ok
- [ ] The fix does not change a customer's website in any surprising or unintended way
  - [ ] If Divi's design <em>is</em> adjusted, settings are migrated gracefully and invisibly

<strong>Design Review</strong>
- [ ] UI is to spec
- [ ] UX is to spec
- [ ] Design is to spec
- [ ] No surprising visual changes to a customer's website will be experienced after updating
  - [ ] Any/all visual changes are migrated gracefully

<strong>Dev Review</strong>
- [ ] Test(s) included and coverage acceptable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated? 
- [ ] i18n used if applicable
- [ ] Implementation acceptable, DRY, Scaleable, future safe
- [ ] Best practices used, cleverness and grokage, ES6 use, etc.

**Comment by Paul-S-Taylor** [comment]
Agree. It would be very useful to have some JSX examples from the original modules.

**Comment by vHeemstra** [comment]
I agree !
If the original JSX files of the existing modules would be provided, all Divi developers could more easily take that as a great starting point for customizing it to their own version/module.

In addition: it could be a starting point for a Divi developers platform where people can upload/share their own built modules for others to use as well.

**Comment by thelanj** [comment]
+1 to giving us the JSX of built in modules...frustratingly hard to build a simple blog type module for custom post types.

**Comment by titantwentyone** [comment]
This is still an open issue. Is there any word from ET regarding this? Extending Divi feels overly complicated for small additions to existing modules.

**Comment by onsterion** [comment]
I really need this for extend the Shop Module.


Any news about this?



Regards,
Lucas - CTO - Kubey Machine.
